content/articles/how-to-implement-zero-shot-classification-using-python/index.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /how-to-implement-zero-shot-classification-using-python/
+title: How to Implement Zero-Shot Classification using Python
+description: This tutorial will leverage the zero-shot classification model from Hugging Face to extract model predictions and perform multi-class classification. 
+author: esther-awuor
+date: 2022-03-11T00:00:00-12:00
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/how-to-implement-zero-shot-classification-using-python/hero.png
+    alt: How to Implement Zero-Shot Classification using Python Example Image
+---
+Machine learning is an ever-evolving field. One area of machine learning that has greatly evolved over a few years is Natural Language Processing (NLP). 
+<!--more-->
+The [HuggingFace](https://huggingface.co/) organization has been at the forefront in making contributions in this field. This tutorial will leverage the zero-shot classification model from Hugging Face to extract model predictions and perform multi-class classification. 
+
+### Prerequisites
+To follow along with this tutorial, you need o be familar with:
+- Machine Learning modeling.
+- Natural Language processing.
+
+### Outline
+- [Zero-shot classification](#zero-shot-classification)
+- [How to leverage the Hugging Face API to perform classification](how-to-leverage-the-hugging-face-api-to-perform-classification)
+- [Performing multi-class classification](#performing-multi-class-classification)
+- [Wrapping-up](#wrapping-up)
+- [Further reading](#further-reading)
+
+### Zero-shot classification
+Zero-shot classification is a technique that allows us to associate an appropriate label with a piece of text. This association is irrespective of the text domain and the aspect. For example, it can be a  topic, emotion, or event described by the label. To perform zero-shot classification, we need a zero-shot model.
+
+A zero-shot model allows us to classify data that had not been previously used to build the model. In simple terms, it is the use of a model built by other people, against your data. Hugging Face allows us to leverage this pre-trained model to perform classification using a few lines of code.
+
+### How to leverage the Hugging Face API to perform classification
+Let's begin by installing the main dependency, Transformers.
+
+```bash
+!pip install transformers
+```
+We need to import `pipeline` into our code.
+
+```python
+from transformers import pipeline
+```
+This next step involves initializing our pipeline. We save the result inside a variable called `classifier_pipeline`. We are also using the `facebook/bart-large-mnli` model. There are other zero-shot models available. We are using this model because it is the most popular with over one million downloads. Please refer to this [link](https://huggingface.co/models?pipeline_tag=zero-shot-classification&sort=downloads) to access the other models.
```

Comment:
```suggestion
This next step involves initializing our pipeline. We save the result inside a variable called `classifier_pipeline`. We are also using the `facebook/bart-large-mnli` model. There are other zero-shot models available. We use this model because it is the most popular, with over one million downloads. Please refer to this [link](https://huggingface.co/models?pipeline_tag=zero-shot-classification&sort=downloads) to access the other models.
```